homeassistant/components/motioneye/__init__.py

```diff
@@ -0,0 +1,263 @@
+
+import asyncio
+import logging
+from typing import Any, Callable
+
+from motioneye_client.client import (
+    MotionEyeClient,
+    MotionEyeClientError,
+    MotionEyeClientInvalidAuthError,
+)
+from motioneye_client.const import KEY_CAMERAS, KEY_ID, KEY_NAME
+
+from homeassistant.components.camera.const import DOMAIN as CAMERA_DOMAIN
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigEntry
+from homeassistant.const import CONF_SOURCE, CONF_URL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_ADMIN_PASSWORD,
+    CONF_ADMIN_USERNAME,
+    CONF_CLIENT,
+    CONF_CONFIG_ENTRY,
+    CONF_COORDINATOR,
+    CONF_ON_UNLOAD,
+    CONF_SURVEILLANCE_PASSWORD,
+    CONF_SURVEILLANCE_USERNAME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    MOTIONEYE_MANUFACTURER,
+    SIGNAL_CAMERA_ADD,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [CAMERA_DOMAIN]
+
+
+def create_motioneye_client(
+    *args: Any,
+    **kwargs: Any,
+) -> MotionEyeClient:
+    """Create a MotionEyeClient."""
+    return MotionEyeClient(*args, **kwargs)
+
+
+def get_motioneye_device_unique_id(config_entry_id: str, camera_id: int) -> str:
+    """Get the unique_id for a motionEye device."""
+    return f"{config_entry_id}_{camera_id}"
+
+
+def get_motioneye_entity_unique_id(
+    config_entry_id: str, camera_id: int, entity_type: str
+) -> str:
+    """Get the unique_id for a motionEye entity."""
+    return f"{config_entry_id}_{camera_id}_{entity_type}"
+
+
+def get_camera_from_cameras(
+    camera_id: int, data: dict[str, Any]
+) -> dict[str, Any] | None:
+    """Get an individual camera dict from a multiple cameras data response."""
+    for camera in data.get(KEY_CAMERAS) or []:
+        if camera.get(KEY_ID) == camera_id:
+            val: dict[str, Any] = camera
+            return val
+    return None
+
+
+def is_acceptable_camera(camera: dict[str, Any] | None) -> bool:
+    """Determine if a camera dict is acceptable."""
+    return bool(camera and KEY_ID in camera and KEY_NAME in camera)
+
+
+@callback
+def listen_for_new_cameras(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    add_func: Callable,
+) -> None:
+    """Listen for new cameras."""
+
+    hass.data[DOMAIN][entry.entry_id][CONF_ON_UNLOAD].extend(
+        [
+            async_dispatcher_connect(
+                hass,
+                SIGNAL_CAMERA_ADD.format(entry.entry_id),
+                add_func,
+            ),
+        ]
+    )
+
+
+async def async_setup(hass: HomeAssistant, config: dict[str, Any]) -> bool:
+    """Set up the motionEye component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
```

Comment:
This is no longer necessary. Instead add `hass.data.setdefault(DOMAIN, {})` to async_setup_entry.
```suggestion
```